sonar-update-center-common/src/test/java/org/sonar/updatecenter/common/SonarTest.java

```diff
@@ -44,6 +44,21 @@ public void should_not_get_undeclared_releases() {
     sonar.getRelease(Version.create("3.3"));
   }
 
+  @Test
+  public void getReleaseWithProduct_shouldReturnReleaseByProduct() {
+    Sonar sonar = new Sonar();
+    SonarRelease paidRelease = new SonarRelease(sonar, Version.create("3.0"), Product.SONARQUBE_SERVER);
+    SonarRelease communityRelease = new SonarRelease(sonar, Version.create("4.0"), Product.SONARQUBE_COMMUNITY_BUILD);
+    SonarRelease oldRelease = new SonarRelease(sonar, Version.create("5.0"), Product.OLD_SONARQUBE);
+    sonar.addRelease(paidRelease);
+    sonar.addRelease(communityRelease);
+    sonar.addRelease(oldRelease);
```

Comment:
Adding another release with same version and different products breaks the test.
Is that a legit case? Added a comment on the SonarRelease class about it.

Reply:
I went back and forth with it, even wanted to add the test to express that we don't want to have same version for two different products but since `addRelease` method is used also by plugins I don't want to take a risk here and throw exception when this happens.
I think it is the best to leave it as it is.